lib/trino-parquet/src/main/java/io/trino/parquet/ParquetTypeUtils.java

```diff
@@ -299,14 +299,14 @@ private static Optional<Field> constructField(Type type, ColumnIO columnIO, bool
             if (!(columnIO instanceof GroupColumnIO groupColumnIo)) {
                 throw new IllegalStateException("Expected columnIO to be GroupColumnIO but got %s".formatted(columnIO.getClass().getSimpleName()));
             }
-            Field valueField = constructField(VARBINARY, groupColumnIo.getChild(0), false).orElseThrow();
+            PrimitiveField valueField = (PrimitiveField) constructField(VARBINARY, groupColumnIo.getChild(0), false).orElseThrow();
             PrimitiveField metadataField = (PrimitiveField) constructField(VARBINARY, groupColumnIo.getChild(1), false).orElseThrow();
             return Optional.of(new VariantField(
                     type,
                     repetitionLevel,
                     definitionLevel,
                     required,
-                    valueField,
+                    new PrimitiveField(valueField.getType(), false, valueField.getDescriptor(), valueField.getId()),
```

Comment:
Does it mean Parquet spec wrong? 
https://github.com/apache/parquet-format/blob/master/VariantEncoding.md#variant-in-parquet

> The value field must be annotated as **_required_** for unshredded Variant values